backend/src/apiserver/main.go

```diff
@@ -235,10 +267,42 @@ func startHttpProxy(resourceManager *resource.ResourceManager) {
 	// Register a handler for Prometheus to poll.
 	topMux.Handle("/metrics", promhttp.Handler())
 
+	go startWebhookHTTPSProxy(controllerClient)
```

Comment:
We should make it so that when the below call of `http.ListenAndServe` closes, the webhook HTTP server shuts down gracefully. See this as an example:
https://stackoverflow.com/a/42533360

`startWebhookHTTPSProxy` could be run not as a goroutine, launch the `ListenAndServe` call to a goroutine, and return the HTTP server to this function to defer the graceful shutdown.